src/components/v2/appDetails/sourceInfo/EnvironmentSelector.component.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useState } from 'react';
+import Select, { components } from 'react-select';
+import { multiSelectStyles, SingleSelectOption as Option } from '../../../common';
+import ScalePodModalComponent from './ScalePodModal.component';
+import './sourceInfo.css';
+import IndexStore from '../index.store';
+import { AppEnvironment } from './environment.type';
+import { useParams, useHistory, useRouteMatch, generatePath, Route, useLocation } from 'react-router';
+
+import { getAppOtherEnvironment } from '../appDetails.api';
+import { useSharedState } from '../../utils/useSharedState';
+
+function EnvironmentSelectorComponent() {
+    const params = useParams<{ appId: string; envId?: string }>();
+    const { url, path } = useRouteMatch();
+    const history = useHistory();
+
+    const [showhiberbateConfirmationModal, setshowHibernateConfirmationModal] = useState(false);
+
+    const [environments, setEnvironments] = useState<Array<AppEnvironment>>();
+
+    const [appDetails] = useSharedState(IndexStore.getAppDetails(), IndexStore.getAppDetailsObservable());
+
+    useEffect(() => {
+        if (params.appId) {
+            getAppOtherEnvironment(params.appId)
+                .then((response) => {
+                    setEnvironments(response.result || []);
+                })
+                .catch((error) => {
+                    console.error('erroe in fetching environments');
+                    setEnvironments([]);
+                });
+        }
+    }, [params.appId]);
+
+    useEffect(() => {
+        if (!params.envId && appDetails.environmentId) {
+            handleEnvironmentChange(appDetails.environmentId);
+        }
+    }, [appDetails.environmentId]);
+
+    const handleEnvironmentChange = (envId: number) => {
+        history.push(`${url}/${envId}`);
```

Comment:
this ll result in wrong url

Reply:
This behaviour is only true in case of devtron app. (Not in the scope of Charts)